diesel/src/types/mod.rs

```diff
@@ -337,18 +338,109 @@ pub enum IsNull {
     No,
 }
 
+#[derive(Clone, Copy)]
+#[doc(hidden)]
+pub struct ToSqlOutput<'a, T, DB> where
+    DB: TypeMetadata,
+    DB::MetadataLookup: 'a,
+{
+    out: T,
+    metadata_lookup: &'a DB::MetadataLookup,
+}
+
+impl<'a, T, DB: TypeMetadata> ToSqlOutput<'a, T, DB> {
+    pub fn new(out: T, metadata_lookup: &'a DB::MetadataLookup) -> Self {
+        ToSqlOutput { out, metadata_lookup }
+    }
+
+    pub fn with_buffer<U>(&self, new_out: U) -> ToSqlOutput<'a, U, DB> {
+        ToSqlOutput {
+            out: new_out,
+            metadata_lookup: self.metadata_lookup,
```

Comment:
Future-proof this even more against diff churn with `... self }` instead of listing the other fields (like `metadata_lookup` here)